### PR TITLE
Add versioned labels to images

### PR DESCRIPTION
**Pull request by alculquicondor**
Base and test images should include the version as part of the label

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -67,7 +67,9 @@ test: bin/envtest scheduler-plugins-crd
 
 # Only works with CONTROLLER_VERSION=v2
 .PHONY: test_e2e
-test_e2e: export TEST_MPI_OPERATOR_IMAGE = ${IMAGE_NAME}:${RELEASE_VERSION}
+test_e2e: export TEST_MPI_OPERATOR_IMAGE=${IMAGE_NAME}:${RELEASE_VERSION}
+test_e2e: export TEST_OPENMPI_IMAGE=mpioperator/mpi-pi:${RELEASE_VERSION}-openmpi
+test_e2e: export TEST_INTELMPI_IMAGE=mpioperator/mpi-pi:${RELEASE_VERSION}-intel
 test_e2e: bin/kubectl kind helm images test_images dev_manifest scheduler-plugins-chart
 	SCHEDULER_PLUGINS_VERSION=$(SCHEDULER_PLUGINS_VERSION) go test -v ./test/e2e/...
```
Comment (tenzen-y) [inline Makefile:74]:
```suggestion
test_e2e: export TEST_INTELMPI_IMAGE=mpioperator/mpi-pi:${RELEASE_VERSION}-intel
test_e2e: export SCHEDULER_PLUGINS_VERSION=${SCHEDULER_PLUGINS_VERSION}
test_e2e: bin/kubectl kind helm images test_images dev_manifest scheduler-plugins-chart
	go test -v ./test/e2e/...
```

Can we use the same way to set up all env vars?

File: Makefile [hunk 2]
```diff
@@ -94,17 +96,17 @@ clean:
...
 .PHONY: test_images
 test_images:
-	${IMG_BUILDER} build --platform $(PLATFORMS) --build-arg port=${BASE_IMAGE_SSH_PORT} -t mpioperator/base build/base
-	${IMG_BUILDER} build --platform $(PLATFORMS) -t mpioperator/openmpi build/base -f build/base/openmpi.Dockerfile
-	${IMG_BUILDER} build --platform $(PLATFORMS) -t mpioperator/openmpi-builder build/base -f build/base/openmpi-builder.Dockerfile
-	${IMG_BUILDER} build --platform $(PLATFORMS) -t mpioperator/mpi-pi:openmpi examples/v2beta1/pi
-	${IMG_BUILDER} build --platform $(PLATFORMS) -t mpioperator/intel build/base -f build/base/intel.Dockerfile
-	${IMG_BUILDER} build --platform $(PLATFORMS) -t mpioperator/intel-builder build/base -f build/base/intel-builder.Dockerfile
-	${IMG_BUILDER} build --platform $(PLATFORMS) -t mpioperator/mpi-pi:intel examples/v2beta1/pi -f examples/v2beta1/pi/intel.Dockerfile
+	${IMG_BUILDER} build $(BUILD_ARGS) --platform $(PLATFORMS) --build-arg port=${BASE_IMAGE_SSH_PORT} -t mpioperator/base:${RELEASE_VERSION} build/base
```
Comment (tenzen-y) [inline Makefile:103]: Do we actually use `BUILD_ARGS`? It doesn't seem to use that env vars in the whole Makefile.

**Review by tenzen-y — COMMENTED** [review]
@alculquicondor Overrolls looks good to me.
I left a few comments for some nits.

File: Makefile (hunk 2 above)
Comment (alculquicondor) [reply Makefile:103]: They can be used to directly push like this: `docker buildx build --push`.

File: Makefile (hunk 1 above)
Comment (alculquicondor) [reply Makefile]: done

File: Makefile (hunk 2 above)
Comment (tenzen-y) [reply Makefile:104]: Ah, I see. Thanks for clarifying!

File: Makefile (hunk 1 above)
Comment (alculquicondor) [reply Makefile]: It fails

```
Run make RELEASE_VERSION=test CONTROLLER_VERSION=v2 test_e2e
Makefile:73: *** Recursive variable 'SCHEDULER_PLUGINS_VERSION' references itself (eventually).  Stop.
mkdir -p /home/runner/work/mpi-operator/mpi-operator/bin
```

Perhaps it's not actually necessary to add

Comment (tenzen-y) [reply Makefile]: You're right.

**Comment by tenzen-y** [comment]
Why an `approved` label isn't added to this PR 🤔

**Comment by alculquicondor** [comment]
/approve
?

**Comment by tenzen-y** [comment]
I wonder why a PR creator can add approved labels to their PR.

**Comment by alculquicondor** [comment]
> I wonder why a PR creator can add approved labels to their PR.

Only because I'm an approver. It works the same way in k8s

**Comment by tenzen-y** [comment]
> > I wonder why a PR creator can add approved labels to their PR.
> 
> Only because I'm an approver. It works the same way in k8s

Ah, I see.

**Pull request merged**